src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -412,6 +449,8 @@ def _visit_deref(self, node: gtir.FunCall) -> ValueExpr:
 
     def _visit_neighbors(self, node: gtir.FunCall) -> DataExpr:
         assert len(node.args) == 2
+        assert isinstance(node.type, itir_ts.ListType)
+        assert self.reduce_identity is not None
```

Comment:
My question was not that well phrased.
In `translate_as_fieldop()` you wrote "[...] when the argument visitor hits an expression returning a neighbors list (see the second if-branch), we stop carrying the reduce identity further [...]".
So my question was aimed at, why are you requiring that here the reduce expression is not `None`.

Reply:
Because here we are visiting the neighbor expression itself. This is not the argument visitor (by that I mean we are not visiting the input arguments to the neighbor expression).